src/tiered/tiered_opener.c

```diff
@@ -0,0 +1,99 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __tiered_opener_open --
+ *     Open an object by number.
+ */
+static int
+__tiered_opener_open(WT_BLOCK_FILE_OPENER *opener, WT_SESSION_IMPL *session, uint64_t object_id,
+  WT_FS_OPEN_FILE_TYPE type, u_int flags, WT_FH **fhp)
+{
+    WT_BUCKET_STORAGE *bstorage;
+    WT_DECL_RET;
+    WT_TIERED *tiered;
+    const char *object_name, *object_uri;
+
+    tiered = opener->cookie;
+    object_uri = NULL;
+
+    WT_ASSERT(session,
+      (object_id > 0 && object_id <= tiered->current_id) || object_id == WT_TIERED_TOP_OBJECT_ID);
+    /*
+     * FIXME-WT-7590 we will need some kind of locking while we're looking at the tiered structure.
+     * This can be called at any time, because we are opening the objects lazily.
+     */
+    if (object_id == tiered->current_id || object_id == WT_TIERED_TOP_OBJECT_ID) {
+        bstorage = NULL;
+        object_name = tiered->tiers[WT_TIERED_INDEX_LOCAL].name;
+        if (!WT_PREFIX_SKIP(object_name, "file:"))
+            WT_RET_MSG(session, EINVAL, "expected a 'file:' URI");
+    } else {
+        WT_ERR(
+          __wt_tiered_name(session, &tiered->iface, object_id, WT_TIERED_NAME_OBJECT, &object_uri));
```

Comment:
The `wt_tiered_name` function could take a (new) `WT_TIERED_NAME_ONLY` flag perhaps (or `WT_TIERED_NAME_NO_PREFIX`). Callers in `conn_tiered.c:wt_tiered_do_flush` are also using it in this manner to the the URI and then stripping off the prefix. That could be another ticket, just pointing it out.

Reply:
I created WT-7635 to track this.